source/speech.py

```diff
@@ -626,7 +640,21 @@ def speakTypedCharacters(ch):
 			log.io("typed word: %s"%typedWord)
 		if config.conf["keyboard"]["speakTypedWords"] and not typingIsProtected:
 			speakText(typedWord)
-	if config.conf["keyboard"]["speakTypedCharacters"] and ord(ch)>=32:
+	global _suppressSpeakTypedCharactersData
+	if _suppressSpeakTypedCharactersData:
+		number, supTime = _suppressSpeakTypedCharactersData
+		# We primarily suppress based on character count,
+		# but time out after a short while just in case.
+		timeOk = time.time() - supTime <= 0.1
+		suppress = number > 0 and timeOk
+		number -= 1
+		if number > 0 and timeOk:
+			_suppressSpeakTypedCharactersData = (number, supTime)
+		else:
+			_suppressSpeakTypedCharactersData = None
+	else:
+		suppress = False
+	if not suppress and config.conf["keyboard"]["speakTypedCharacters"] and ord(ch)>=32:

```

Comment:
Quite true. However, as that's very much unrelated code that existed previously (I just added an extra condition), I'd prefer to address this somewhere else.